gpt-4.1 implement-phase promise-loop hallucination — both 0x-premium copilot models now broken

## Problem

After PR#644 re-applied `med: gpt-4.1` (loop 309, 2026-04-19T19:04Z), implement-harness vessels on gpt-4.1 hit a **promise-loop hallucination** in the `implement` phase:

- `analyze` + `plan` phases complete cleanly (gpt-4.1 writes real structured output).
- `implement` phase: model describes the plan, then repeats escalating promises without ever calling a tool:
  - "I'll begin by..."
  - "Proceeding with the actual file changes now."
  - "Acknowledged. I am now making the required code and test changes."
- After ~15-25 min of self-promising, the model drifts and starts asking "No user task has been specified. Please provide a specific request..."
- Harness SIGKILLs at 45 min wall-clock (`error: phase implement: signal: killed`).

Observed on 4 vessels: `issue-647-retry-2`, `issue-653-retry-2-retry-1` (both started 19:49:13Z, both SIGKILL'd at 20:34Z), and in-flight at time of filing: `issue-661-retry-3`, `issue-650-retry-4`.

Evidence: `.xylem/state/phases/issue-647-retry-2/implement.output` (head = real plan restatement, tail = "no user task" starvation). `failure-review.json` has `recovery_action: human_escalation`, `retry_suppressed: true`.

## Why this is critical

**Scanner chain-retry (PR#660) bypasses `retry_suppressed`.** Every SIGKILL'd vessel gets a `-retry-N+1` child on the next scan, which burns another 45 min of the 0x-premium copilot quota and daemon concurrency. Cascade is unbounded as long as the issue keeps its `ready-for-work` label.

Blocks: #647 (invariant-coverage CI), #649 (naive-reference test), #650 (I1/I3/I6/I8/I11/I14 property tests), #653 (implement-feature post-pr gate), #661 (scanner.go:192 doc-comment), #662 (chain-retry depth cap).

## Suggested fixes (ranked)

1. **Pin `med: claude-opus-4-6` for `implement-harness` workflow** (`.xylem.yml` — route implement phase to claude even when tier=med). Burns Max credits but unblocks the cascade. Narrow scope: only the `implement` phase of `implement-harness`, not `fix-bug` / `implement-feature` / `fix-pr-checks`.
2. **Prompt anchor in `.xylem/prompts/implement-harness/implement.md`**: prepend "Your FIRST action MUST be a concrete tool call (Read or Grep). Do not describe or narrate; act." Forces gpt-4.1 out of promise mode.
3. **Add scanner chain-retry depth cap** (hnipps flagged this in the PR#660 review — filed as #661 doc-only, but actual depth guard is worth the 10 LOC). Stops unbounded burn.
4. **Stopgap**: set `max_turns: 20` for `implement-harness` workflow — limits burn per vessel from 45 min to ~5 min when the model loops.

## Reproduction

```bash
cd /Users/harry.nicholls/repos/xylem/.daemon-root
# Any implement-harness vessel on med tier with fresh gpt-4.1 subprocess:
head -40 .xylem/state/phases/issue-647-retry-2/implement.output   # promise loop
tail -20 .xylem/state/phases/issue-647-retry-2/implement.output   # "no user task" drift
cat .xylem/state/phases/issue-647-retry-2/failure-review.json | grep -E 'error|recovery_action|retry_suppressed'
```

Supervisor loop 311 @ 2026-04-19T21:01Z documented the verdict for loop 309's hypothesis (PR#644 gpt-4.1 restore).

**xylem — failed at phase `implement`**

**Error:** gate failed, retries exhausted

<details>
<summary>Gate output (click to expand)</summary>

make: *** No rule to make target `test'.  Stop.


</details>

--- a/.xylem/prompts/implement-harness/implement.md
+++ b/.xylem/prompts/implement-harness/implement.md
@@ -1,3 +1,5 @@
+**Your FIRST action MUST be a concrete tool call (Read, Grep, or Edit). Do not describe, narrate, or restate the plan — act immediately.**
+
 Implement the harness spec step according to the plan.
 
 Issue: {{.Issue.Title}}

--- a/.xylem/workflows/implement-harness.yaml
+++ b/.xylem/workflows/implement-harness.yaml
@@ -14,7 +14,7 @@ phases:
   - name: implement
     prompt_file: .xylem/prompts/implement-harness/implement.md
     max_turns: 80
-    tier: med
+    tier: high
     gate:
       type: command
       run: "cd cli && go vet ./... && go build ./cmd/xylem && go test ./..."

